docs/tutorial/values.md

```diff
@@ -0,0 +1,215 @@
+(end: (x: -5, y: 3), start: (x: 1, y: 2))
+```
+
+Note that tuple names are unordered. That is, `(x: 1, y: 2)` and `(y: 2, x: 1)`
+represent the same tuple. However, for convenience and consistency, arr.ai will
+usually print tuples with names ordered lexicograpically.
+
+Attribute names can be pretty much anything. Names that don't fit the standard
+identifier rules can be written using string syntax:
+
+```text
+@> ('-1-': 23, '-2-': 34)
+('-1-': 23, '-2-': 34)
+@> ('\007': 100)
+('\a': 100)
+@> ('🐭': "mouse", '🐴': "horse", '🐱': "cat")
+('🐭': 'mouse', '🐱': 'cat', '🐴': 'horse')
+```
+
+**Explore:** What do the following output?
+
+```text
+@> ((a: 1, b: 2))
+@> ((()))
+```
+
+## Sets
+
+Sets hold values. Any given value is either in a set or it isn't. There is no
+notion of multiplicity; a value cannot be present more than once. There is also
+no notion of ordering.
+
+```text
+@> {}
+{}
+@> {1, 2, 3}
+{1, 2, 3}
+@> {{1, 2}, {3, 4}}
+{{1, 2}, {3, 4}}
+@> {1, {2, 3}}
+{1, {2, 3}}
+@> {(x: 1), (s: {2, 3})}
+{(s: {2, 3}), (x: 1)}
+@> {()}
+{()}
+@> 3 < 4
+{()}
+```
+
+Note that we introduced a Boolean test in that last expression, `3 < 4`. Since
+`3` is in fact less than `4`, that expression evaluates to `true`. However,
+since there is no Boolean type in arr.ai, `true` is actually an alias for the
+set with just the empty tuple in it, `{()}`.
+
+**Explore:** What do the following output?
+
+```text
+@> true
+@> false
+```
+
+## Special values
+
+It probably feels like the above type system just isn't enough. In real world
+programming, we typically want to work with a range of usefule types, such as
+strings, arrays and functions. These and more are in fact available in arr.ai,
+but they do not exist as distinct types from the above. Rather, they are
+expressed as special compositions of the above types.
+
+### Mapping types
+
+The following sample run demonstrates the syntax for arrays, strings and
+dictionaries. It also shows that they are nothing more than syntactic sugar for
+special compositions of numbers, tuples and sets.
+
+```text
+@> [3, 9, 27]
+[3, 9, 27]
+@> {(@: 0, @item: 3), (@: 1, @item: 9), (@: 2, @item: 27)}
+[3, 9, 27]
+@> "hello"
+'hello'
+@> 'hello'
+'hello'
+@> `hello`
+'hello'
+@> {(@: 0, @char: 104), (@: 1, @char: 101), (@: 2, @char: 108), (@: 3, @char: 108), (@: 4, @char: 111)}
+'hello'
+@> {1: "hi", 2: "bye"}
+{1: 'hi', 2: 'bye'}
+@> {(@: 1, @value: 'hi'), (@: 2, @value: 'bye')}
+{1: 'hi', 2: 'bye'}
+@> {[1, 2]: 42, [3, 5]: 54}
+{[1, 2]: 42, [3, 5]: 54}
+```
+
+**Explore:** How would you describe the following values?
+
+```text
+@> {{}: {}, (): ()}
+@> {{{(x:1): 42}: 54}}
+```
+
+**Food for thought:** Since tuples and dictionaries are both basically key/value
+collections, how would you characterise their differences and can you explain
+why both are available in arr.ai? The following idiomatic example might help.
+
+```text
+@> {"dog": (legs: 4, sound: "bark"), "cat": (legs: 4, sound: "meow")}
```

Comment:
Yep, the key/value concept in sets and tuples also confused me and I'm not sure if the example might be clear enough. I can replace the tuples with sets and they'll be nearly identical.